src/proxy/inbound_passthrough.rs

```diff
@@ -69,13 +76,43 @@ impl InboundPassthrough {
     }
 
     async fn proxy_inbound_plaintext(
+        pi: ProxyInputs,
         source: SocketAddr,
-        inbound: &mut TcpStream,
+        mut inbound: TcpStream,
     ) -> Result<(), Error> {
-        let orig = socket::orig_dst_addr_or_default(inbound);
+        let orig = socket::orig_dst_addr_or_default(&inbound);
+        let Some(upstream) = pi.workloads.fetch_workload(&orig.ip()).await else {
+            return Err(Error::UnknownDestination(orig.ip()))
+        };
+        if !upstream.waypoint_addresses.is_empty() {
+            // This is an inbound request not over HBONE, but we have a waypoint.
+            // The request needs to go through the waypoint for policy enforcement.
```

Comment:
Will L4 traffic goes to waypoint here？

Reply:
Then could i understand that a single Authz policy can be enforced by both waypoint and ztunnel?

Reply:
Yes, but not on a single connection. A given connection will be enforced at waypoint or ztunnel, not both.

This is important because a policy like "allow GET from spiffe://foo" will always deny at ztunnel since we cannot parse L7

Reply:
>Yes, I'd they have one. Note waypoint is never intended to only handle L7 policies. There are plenty of configs that are L4 that only waypoint can handle.

I am concerned for traffic to some L4 ports, it is redirected to waypoint. Why in this case ztunnel can not handle it? Imaging if there are no waypoint like below processing, so I mean could separate the traffic with protocol, not only by ip

Reply:
Understood that.

But as i know, WASM, Ext Auth are mostly L7. We can get the protocol from port name before. If we must support protocol sniffer, this comment can be ignored.